plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -377,24 +377,36 @@ public synchronized void setDatabaseOwner(HiveIdentity identity, String source,
     // TODO: Allow updating statistics for 2 tables in the same transaction
     public synchronized void setTableStatistics(HiveIdentity identity, Table table, PartitionStatistics tableStatistics)
     {
-        AcidTransaction transaction = currentHiveTransaction.isPresent() ? currentHiveTransaction.get().getTransaction() : NO_ACID_TRANSACTION;
+        AcidTransaction transaction = getTransactionForStatisticsUpdate();
```

Comment:
I don't understand the change here. What would happen if
```
AcidTransaction transaction = currentHiveTransaction.isPresent() ? currentHiveTransaction.get().getTransaction() : NO_ACID_TRANSACTION;
```

was retained unmodified?

Reply:
As of today, we should update stats under two circumstances:

1. The table is not transactional.
2. The table is transactional, and the operation is ANALYZE

The helper method getTransactionForStatisticsUpdate() is a central point at which we can make that distinction.

I think it's straightforward to add support for saving stats for INSERT, DELETE and UPDATE as well, but it seemed that should be a separate PR.

Reply:
Thanks for expaining, but i still would want to know what would happen if

```
AcidTransaction transaction = currentHiveTransaction.isPresent() ? currentHiveTransaction.get().getTransaction() : NO_ACID_TRANSACTION;
```

was retained unmodified?

BTW, the decision not to collect stats for transactional INSERT/DELETE/UPDATE should be done earlier, in `io.trino.plugin.hive.HiveMetadata#getStatisticsCollectionMetadataForWrite`. There is no point collecting stats if we know upfront we are not going to save them.

Reply:
> if ... was retained unmodified?

The value-add of the helper method is to ensure that we don't try to update statistics for transactional tables if we know that there are no statistics because the transaction isn't ANALYZE.  I guess I don't understand what needs clarification.

Reply:
I guess i don't understand the problem this is solving. 

If the original code is added back here, which test(s) will fail?

Reply:
Well, I ran `TestHiveTransactionalTable` and `TestHiveBasicStatistics` without the change, and all tests passed.  So I kept the new method `getTransactionForStatisticsUpdate()`, but reverted the change to test for an `ANALYZE` transaction:

```
    private AcidTransaction getTransactionForStatisticsUpdate()
    {
        return currentHiveTransaction.isPresent() ? currentHiveTransaction.get().getTransaction() : NO_ACID_TRANSACTION;
    }
```